libraries/inverse/hpiFit/sensorset.cpp

```diff
@@ -0,0 +1,151 @@
+ * PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ *
+ * @brief    SensorSet class definition.
+ *
+ */
+
+//=============================================================================================================
+// INCLUDES
+//=============================================================================================================
+
+#include <inverse/hpiFit/sensorset.h>
+#include <iostream>
+#include <fwd/fwd_coil_set.h>
+#include <fiff/fiff_ch_info.h>
+
+//=============================================================================================================
+// QT INCLUDES
+//=============================================================================================================
+
+#include <QCoreApplication>
+
+//=============================================================================================================
+// EIGEN INCLUDES
+//=============================================================================================================
+
+#include <Eigen/Dense>
+
+//=============================================================================================================
+// USED NAMESPACES
+//=============================================================================================================
+
+using namespace INVERSELIB;
+using namespace FIFFLIB;
+using namespace FWDLIB;
+using namespace Eigen;
+
+//=============================================================================================================
+// DEFINE GLOBAL METHODS
+//=============================================================================================================
+
+//=============================================================================================================
+// DEFINE MEMBER METHODS
+//=============================================================================================================
+
+SensorSet::SensorSet(const FwdCoilSet::SPtr pFwdSensorSet)
+{
+    if(pFwdSensorSet!=nullptr) {
+        convertFromFwdCoilSet(pFwdSensorSet);
+    }
+}
+
+//=============================================================================================================
+
+void SensorSet::convertFromFwdCoilSet(const QSharedPointer<FWDLIB::FwdCoilSet> pFwdSensorSet)
+{
+    m_ncoils = pFwdSensorSet->ncoil;
+    m_np = pFwdSensorSet->coils[0]->np;
+    initMatrices(m_ncoils,m_np);
+
+    // get data froms Fwd Coilset
+    for(int i = 0; i < m_ncoils; i++){
```

Comment:
Maybe use a range based for loop here. looks nice IMHO 
`for(auto* coil : pFwdSensorSet->coils){`

Reply:
I get following error:
invalid range expression of type 'FWDLIB::FwdCoil **'; no viable 'begin' function available

Reply:
ok got it. `coils` is a double pointer and not an array or an object for which begin and end member functions. Please disregard my comment.